alpha/lib/model/accessControl.php

```diff
@@ -266,8 +266,12 @@ public function applyContext(kEntryContextDataResult &$context, accessControlSco
 			$this->setScope($scope);
 
 		$disableCache = false;
-		$isKsAdmin = $this->scope && $this->scope->getKs() && $this->scope->getKs()->isAdmin();
-		
+		$ks = $this->scope ? $this->scope->getKs() : null;
+
+		$isKsAdmin = $ks && $ks->isAdmin();
+		$isKsReachVendor = $ks && ($ks->getRole() === UserRoleId::REACH_VENDOR_ROLE) &&
```

Comment:
cosmetics - 
$isKsReachVendor = $ks && 
                                   $ks->getRole() === UserRoleId::REACH_VENDOR_ROLE &&
			          $ks->getPrivilegeValue(kSessionBase::PRIVILEGE_VIEW) === $this->scope->getEntryId();

Is pretier.